### PR TITLE
Docstrings

### PR DESCRIPTION
**Pull request by moshez**
Added docstrings. Removed pylint docstring error disablement.

### CODE REVIEW

**Comment by moshez** [comment]
Fixes #8

File: src/gather/tests/test_commands.py [hunk 1]
```diff
@@ -12,29 +14,35 @@
...
     return 'main3', args
 
 @EVIL_COMMANDS.register(name='baz')
 def main4(args):
+    """Plugin registering for the other collector"""
     return 'main4', args
 
 @_helper.weird_decorator
 def weird_function():
+    """Plugin using a wrapper to register"""
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: A wrapper, meaning the decorator calls into `register`, or a `Wrapper`?

File: src/gather/tests/test_commands.py [hunk 2]
```diff
@@ -12,29 +14,35 @@
...
     pass
 
 TRANSFORM_COMMANDS = gather.Collector()
 
 @TRANSFORM_COMMANDS.register(transform=gather.Wrapper.glue(5))
 def fooish():
+    """Plugin registering with a transformation"""
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: ~registering~ registered

File: src/gather/tests/test_commands.py [hunk 3]
```diff
@@ -44,61 +52,78 @@ def fooish():
...
 def weird_name3():
+    """One of several commands registering for same name"""
     pass
 
 class CollectorTest(unittest.TestCase):
 
+    """Tests for collecting plugins"""
+
     def test_collecting(self):
+        """Collecting gives only the registered plugins for given collector"""
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: for *a* given collector

File: src/gather/tests/test_commands.py [hunk 4]
```diff
@@ -44,61 +52,78 @@ def fooish():
...
         collected = NICE_COMMANDS.collect()
         self.assertIn('main1', collected)
         self.assertIs(collected['main1'], main1)
         self.assertNotIn('baz', collected)
 
     def test_non_collision(self):
+        """Collecting with same name for different collectors does not collide"""
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: with *the* same name

What's the difference between collisions and conflicts?

File: src/gather/tests/test_commands.py [hunk 5]
```diff
@@ -44,61 +52,78 @@ def fooish():
...
         self.assertIs(nice['bar'], main3)
         self.assertIs(evil['weird_name'], main3)
 
     def test_cross_module_collection(self):
+        """Collection works when plugins are registered in a different module"""
         collected = _helper.WEIRD_COMMANDS.collect()
         self.assertIn('weird_function', collected)
 
     def test_transform(self):
+        """Collecting transformd plugins applies transform on collection"""
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: ~transformd~ transformed

File: src/gather/tests/test_commands.py [hunk 6]
```diff
@@ -44,61 +52,78 @@ def fooish():
...
         collected = TRANSFORM_COMMANDS.collect()
         self.assertIn('fooish', collected)
         res = collected.pop('fooish')
         self.assertIs(res.original, fooish)
         self.assertEquals(res.extra, 5)
 
     def test_one_of_strategy(self):
+        """:code:`one_of` strategy returns one of the registered plugin for name"""
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: _The_ `one_of` strategy... the registered ~plugin~ plugins

Is there a way to indicate that `name` was a parameter?

File: src/gather/tests/test_commands.py [hunk 7]
```diff
@@ -44,61 +52,78 @@ def fooish():
...
         collected = CONFLICTING_COMMANDS.collect()
         weird_name = collected.pop('weird_name')
         self.assertEquals(collected, {})
         self.assertIn(weird_name, (weird_name1, weird_name2, weird_name3))
 
     def test_explicit_one_of_strategy(self):
+        """:code:`one_of` strategy returns one of the registered plugin
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: ...registered ~plugin~ plugins.

File: src/gather/tests/test_commands.py [hunk 8]
```diff
@@ -44,61 +52,78 @@ def fooish():
...
+
+        Test gives strategy explicitly for name when asked explicitly"""
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: This is unclear.  Test uses the _one_of_ strategy when explicitly asked to for `name`'s resolution?

File: src/gather/_helper.py [hunk 9]
```diff
@@ -1,7 +1,9 @@
+"""Wrap a plugin registration in a function"""
 import gather
 
 WEIRD_COMMANDS = gather.Collector(depth=2)
 
 def weird_decorator(func):
```
Comment (moshez) [inline src/gather/_helper.py]: Moved.

File: src/gather/api.py [hunk 10]
```diff
@@ -111,6 +111,7 @@ def main(args):
                 pass
         """
         def callback(scanner, inner_name, objct):
+            """Venusian callback to be called from scan"""
```
Comment (moshez) [inline src/gather/api.py]: Done

File: src/gather/api.py [hunk 11]
```diff
@@ -121,6 +122,7 @@ def callback(scanner, inner_name, objct):
             objct = transform(objct)
             scanner.update(effective_name, objct)
         def ret(func):
```
Comment (moshez) [inline src/gather/api.py]: done

File: src/gather/api.py [hunk 12]
```diff
@@ -208,6 +215,7 @@ def glue(cls, extra):
         of a :code:`register` call.
         """
         def ret(original):
+            """Return a Wrapper with the original and extra"""
```
Comment (moshez) [inline src/gather/api.py]: Added :code:

File: src/gather/example/main.py [hunk 13]
```diff
@@ -24,21 +29,30 @@ def breakfast(_args):
 
 @BREAKFAST.register()
 class Eggs(object):
+    """Eggs plugin for breakfast"""
     def prepare(self):
+        """Scramble the eggs"""
```
Comment (moshez) [inline src/gather/example/main.py]: done

File: src/gather/example/main.py [hunk 14]
```diff
@@ -24,21 +29,30 @@ def breakfast(_args):
...
         print("Scrambling eggs")
     def eat(self):
+        """Devour the eggs"""
```
Comment (moshez) [inline src/gather/example/main.py]: done

File: src/gather/example/main.py [hunk 15]
```diff
@@ -24,21 +29,30 @@ def breakfast(_args):
...
         print("Devouring eggs")
 
 @BREAKFAST.register()
 class Cereal(object):
+    """Cereal plugin for breakfast"""
     def prepare(self):
+        """Mix it with milk"""
```
Comment (moshez) [inline src/gather/example/main.py]: done

File: src/gather/example/main.py [hunk 16]
```diff
@@ -24,21 +29,30 @@ def breakfast(_args):
...
         print("Mixing cereal and milk")
     def eat(self):
+        """Eat it with a spoon"""
```
Comment (moshez) [inline src/gather/example/main.py]: done

File: src/gather/example/main.py [hunk 17]
```diff
@@ -24,21 +29,30 @@ def breakfast(_args):
...
         print("Eating cereal with a spoon")
 
 @BREAKFAST.register()
 class OrangeJuice(object):
+    """OJ plugin for breakfast"""
     def prepare(self):
+        """Squeeze it"""
```
Comment (moshez) [inline src/gather/example/main.py]: done

File: src/gather/example/main.py [hunk 18]
```diff
@@ -24,21 +29,30 @@ def breakfast(_args):
...
         print("Squeezing orange juice")
     def eat(self):
+        """Drink it"""
```
Comment (moshez) [inline src/gather/example/main.py]: done

File: src/gather/example/selftest.py [hunk 19]
```diff
@@ -8,6 +9,7 @@
 
 @cemain.COMMANDS.register()
 def selftest(_args):
+    """Use example plugins to run a self-test"""
```
Comment (moshez) [inline src/gather/example/selftest.py]: done

File: src/gather/tests/test_commands.py [hunk 20]
```diff
@@ -12,29 +14,35 @@
 
 @NICE_COMMANDS.register()
```
Comment (moshez) [inline src/gather/tests/test_commands.py]: Opening a new ticket for that -- #13

File: src/gather/tests/test_commands.py (hunk 1 above)
Comment (moshez) [inline src/gather/tests/test_commands.py]: The former. Will clarify.

File: src/gather/tests/test_commands.py (hunk 4 above)
Comment (moshez) [inline src/gather/tests/test_commands.py]: Collision is when two plugins are registered for the same name. A conflict is when this happens when you collect with the conflict strategy.

File: src/gather/tests/test_commands.py [hunk 21]
```diff
@@ -12,29 +14,35 @@
...
 def main1(args):
+    """Plugin registering as name of function"""
```
Comment (moshez) [inline src/gather/tests/test_commands.py]: done

File: src/gather/tests/test_commands.py [hunk 22]
```diff
@@ -12,29 +14,35 @@
...
     return 'main1', args
 
 @NICE_COMMANDS.register(name='weird_name')
 def main2(args):
+    """Plugin registering as explicit name"""
```
Comment (moshez) [inline src/gather/tests/test_commands.py]: done

File: src/gather/tests/test_commands.py [hunk 23]
```diff
@@ -12,29 +14,35 @@
...
     return 'main2', args
 
 @NICE_COMMANDS.register(name='bar')
 @EVIL_COMMANDS.register(name='weird_name')
 def main3(args):
+    """Plugin registering for two collectors"""
```
Comment (moshez) [inline src/gather/tests/test_commands.py]: done

File: src/gather/tests/test_commands.py [hunk 24]
```diff
@@ -12,29 +14,35 @@
...
     return 'main3', args
 
 @EVIL_COMMANDS.register(name='baz')
 def main4(args):
+    """Plugin registering for the other collector"""
```
Comment (moshez) [inline src/gather/tests/test_commands.py]: done

File: src/gather/tests/test_commands.py (hunk 2 above)
Comment (moshez) [inline src/gather/tests/test_commands.py]: done

File: src/gather/tests/test_commands.py (hunk 3 above)
Comment (moshez) [inline src/gather/tests/test_commands.py]: done

File: src/gather/tests/test_commands.py (hunk 5 above)
Comment (moshez) [inline src/gather/tests/test_commands.py]: done

File: src/gather/tests/test_commands.py (hunk 6 above)
Comment (moshez) [inline src/gather/tests/test_commands.py]: I hope I clarified this

File: src/gather/tests/test_commands.py (hunk 7 above)
Comment (moshez) [inline src/gather/tests/test_commands.py]: done

File: src/gather/tests/test_commands.py (hunk 8 above)
Comment (moshez) [inline src/gather/tests/test_commands.py]: Clarified.

File: src/gather/tests/test_commands.py (hunk 4 above)
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: Do you see an opportunity for unifying these terms?

File: src/gather/api.py [hunk 25]
```diff
@@ -111,6 +111,11 @@ def main(args):
                 pass
         """
         def callback(scanner, inner_name, objct):
+            ("""Venusian_ callback to be called from scan
```
Comment (markrwilliams) [inline src/gather/api.py]: Why `(...)` around the docstring??

File: src/gather/tests/test_commands.py (hunk 23 above)
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: Hm, still says `registering`...

File: src/gather/tests/test_commands.py [hunk 26]
```diff
@@ -108,7 +141,10 @@ def test_conflicting_strategy(self):
 
 class RunTest(unittest.TestCase):
```
Comment (markrwilliams) [inline src/gather/tests/test_commands.py]: Whitespace?

File: src/gather/api.py (hunk 25 above)
Comment (moshez) [inline src/gather/api.py]: So I can break a line.

**Pull request merged**